src/main/java/org/apache/ibatis/migration/operations/UpOperation.java

```diff
@@ -70,13 +71,13 @@ public UpOperation operate(ConnectionProvider connectionProvider, MigrationLoade
       Collections.sort(migrations);
       String skippedOrMissing = checkSkippedOrMissing(changesInDb, migrations);
       int stepCount = 0;
-      ScriptRunner runner = getScriptRunner(connectionProvider, option, printStream);
-
-      Map<String, Object> hookBindings = new HashMap<String, Object>();
 
+      Map<String, Object> hookBindings = new HashMap<>();
+      ScriptRunner runner = null;
```

Comment:
This is a little shady since Line 117 needs the ScriptRunner to run the abort script. I hope this is still fine.

Reply:
I might adjust details later, but it's fine for now.
Thanks a lot!

Reply:
Please let me know how you adjusted it. I'd love to learn.

Reply:
It's just a minor optimization : 672231a197384b42c13d04e0abcb2f3ca4fa88b2

This PR was perfect as it was. :)
I prefer making minimum changes at a time.